spec/lib/danfe_lib/infadic_spec.rb

```diff
@@ -36,65 +31,108 @@
+                <pesoB>1100.000</pesoB>
+              </vol>
+              <vol>
+                <qVol>2</qVol>
+                <esp>VOLUMES 2</esp>
+                <marca>DIVERSOS 2</marca>
+                <nVol>2</nVol>
+                <pesoL>2000.000</pesoL>
+                <pesoB>2200.000</pesoB>
+              </vol>
+              <vol>
+                <qVol>3</qVol>
+                <esp>VOLUMES 3</esp>
+                <marca>DIVERSOS 3</marca>
+                <nVol>3</nVol>
+                <pesoL>3000.000</pesoL>
+                <pesoB>3300.000</pesoB>
+              </vol>
+            </transp>
+            <infAdic>
+              <infCpl>Uma observação</infCpl>
+            </infAdic>
             <ICMSTot>
-              <vFCPUFDest>0.00</vFCPUFDest>
-              <vICMSUFDest>0.00</vICMSUFDest>
-              <vICMSUFRemet>0.00</vICMSUFRemet>
+              <vFCPUFDest>4892.78</vFCPUFDest>
+              <vICMSUFDest>2915.78</vICMSUFDest>
+              <vICMSUFRemet>75394.78</vICMSUFRemet>
             </ICMSTot>
-          </total>
-        </infNFe>
-      </NFe>
-      eos
+          </infNFe>
+        </NFe>
+      EOS
     end
+    let(:volumes_number) { 1 }
 
-    context "without extra volume" do
-      before do
-        subject.render(1)
-        File.delete(output_pdf) if File.exist?(output_pdf)
+    before do
+      subject.render(volumes_number)
+      File.delete(output_pdf) if File.exist?(output_pdf)
+    end
+
+    context 'when there is no information' do
+      let(:xml_as_string) do
+        <<-EOS
+          <NFe xmlns="http://www.portalfiscal.inf.br/nfe">
+            <infNFe Id="NFe25111012345678901234550020000134151000134151" versao="2.00">
+              <transp>
+                <vol>
+                  <qVol>1</qVol>
+                  <esp>VOLUMES 1</esp>
+                  <marca>DIVERSOS 1</marca>
+                  <nVol>1</nVol>
+                  <pesoL>1000.000</pesoL>
+                  <pesoB>1100.000</pesoB>
+                </vol>
+                <vol>
+                  <qVol>2</qVol>
+                  <esp>VOLUMES 2</esp>
+                  <marca>DIVERSOS 2</marca>
+                  <nVol>2</nVol>
+                  <pesoL>2000.000</pesoL>
+                  <pesoB>2200.000</pesoB>
+                </vol>
+                <vol>
+                  <qVol>3</qVol>
+                  <esp>VOLUMES 3</esp>
+                  <marca>DIVERSOS 3</marca>
+                  <nVol>3</nVol>
+                  <pesoL>3000.000</pesoL>
+                  <pesoB>3300.000</pesoB>
+                </vol>
+              </transp>
+              <total>
+                <ICMSTot>
+                  <vFCPUFDest>4892.78</vFCPUFDest>
+                  <vICMSUFRemet>75394.78</vICMSUFRemet>
+                </ICMSTot>
+              </total>
+            </infNFe>
+          </NFe>
+        EOS
       end
 
-      it "renders xml to the pdf" do
+      it 'renders xml to the pdf' do
```

Comment:
Eu melhoraria todos os `its`, por exemplo esse aqui, no contexto fala que não tem informação, logo no `it` deveria expor o que isso causa no danfe. Poderia ser algo como, 'não renderiza as informações xyz'